tests/foreman/maintain/test_clone.py

```diff
@@ -0,0 +1,163 @@
+"""Tests for satellite-maintain clone
+
+:Requirement: foreman-maintain
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: SatelliteClone
+
+:Assignee: lpramuk
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import pytest
+import requests
+
+from robottelo import constants
+from robottelo.config import settings
+
+USER = settings.subscription.rhn_username
+PASSWORD = settings.subscription.rhn_password
+SSH_PASS = settings.server.ssh_password
+POOL = settings.subscription.rhn_poolid
+BACKUP_DIR = '/var/backup'
+
+
+@pytest.mark.parametrize(
+    "sat_ready_rhel", [7, 8] if settings.server.version.rhel_version < 8 else [8], indirect=True
+)
+@pytest.mark.parametrize('backup_type', ['online', 'offline'])
+def test_positive_clone_backup(target_sat, sat_ready_rhel, backup_type):
+    rhel_version = sat_ready_rhel._v_major
+
+    # SATELLITE PART - SOURCE SERVER
+    # Enabling and starting services
+    assert target_sat.execute('satellite-maintain service enable').status == 0
+    assert target_sat.execute('satellite-maintain service start').status == 0
+    # Doing backup
+    backup_result = target_sat.execute(
+        f'satellite-maintain backup {backup_type} --assumeyes /var/backup'
+    )
+    assert backup_result.status == 0
+    # Getting backup directory
+    sat_backup_dir = backup_result.stdout.strip().split()[-2]
+
+    source_sat = target_sat
+    # Copying satellite backup to target RHEL
+    assert (
+        source_sat.execute(
+            f'sshpass -p "{SSH_PASS}" scp -o StrictHostKeyChecking=no -r {sat_backup_dir}'
+            f'root@{sat_ready_rhel.hostname}:/backup/'
+        ).status
+        == 0
+    )
+
+    # RHEL PART - TARGET SERVER
+    assert sat_ready_rhel.execute('ls /backup/config_files.tar.gz').status == 0
+    # Registering to cdn
+    sat_ready_rhel.register_to_cdn()
+    # Disabling repositories
+    assert sat_ready_rhel.execute('subscription-manager repos --disable=*').status == 0
+    # Getting satellite maintenance repo
+    res = requests.get(
+        f'http://ohsnap.sat.engineering.redhat.com/api/releases/{target_sat.version}/'
+        f'el{rhel_version}/satellite/repo_file'
+    )
+    escaped = res.text.replace('$', '\\$')
+    assert sat_ready_rhel.execute(f'echo "{escaped}" > /etc/yum.repos.d/satellite.repo').status == 0
+    assert sat_ready_rhel.execute('ls /etc/yum.repos.d/satellite.repo').status == 0
```

Comment:
We've a helper available for downloading repofiles available in contenthost_mixin
```suggestion
     sat_ready_rhel.download_repofile()
```

Reply:
As I am looking at _download_repofile_ definition it returns only tools not satellite repofile link
https://github.com/SatelliteQE/robottelo/blob/9609c95caedaa4596fb257620bf663845c269f89/robottelo/host_helpers/contenthost_mixins.py#L43-L50

Reply:
@JacobCallahan @jyejare What is your suggestion here? It would be really nice to use this method, but it does not allow specifying the `product` and it relies on the class name instead.

Reply:
correct. dogfood_repofile is an attribute of each host object. you can instead get the repofile attribute from a satellite, then manually download it

Reply:
> correct. dogfood_repofile is an attribute of each host object. you can instead get the repofile attribute from a satellite, then manually download it

This also seems reasonable. But what if there is a test that installs Satellite on a ContentHost without the need of target_sat? Yes, target_sat is always spined up during the pytest session, but I can imagine such a test case for PIT where we want to test Satellite installation using robottelo, not AAP.

Reply:
@JacobCallahan @mshriver I feel we are making this tightly coupled, even if you call ` Satellite(rhel_vm.hostname).download_repofile()` there is no guarantee that we could get a valid one. 

I'm my case https://github.com/SatelliteQE/robottelo/pull/10033/files#diff-6bf5cc432e0611da570dd30c289a1cfca53618276a13f7c1a04f2981b3bc0708R110 even I wanted to get the satellite repos in the RHEL machine as test verify custom cert satellite installation. This generates the incorrect curl URL as `http://<host>/api/releases/upstream/el8/satellite/repo_file` 

If you see the above URL `upstream` is dynamically generated based on the satellite.version property. I will have to fix this using kwargs now in satellite class. Would it be good to have a generic method and provide more power tests to download the repo?

Reply:
+1, I think we should make this helper func more generic to download repos based on product arg passed, and not keep this tightly coupled with Satellite object.
With introduction to new repo API's in ohsnap, we do've more product options like satellite, capsule, client, and utils, so I feel it would be good to handle those products too in this helper, not something blocking this PR 

@JacobCallahan @mshriver any thoughts?

Reply:
@omkarkhatavkar @Gauravtalreja1 i'm certainly open to the property becoming a method, providing more control.

Reply:
> @omkarkhatavkar @Gauravtalreja1 i'm certainly open to the property becoming a method, providing more control.

@JacobCallahan Is that something that you would like to work on as an extension of the work that you have done in #9849?